ros2/colcon.pkg

```diff
@@ -1,5 +1,5 @@
 {
     "name": "is-ros2",
     "type": "ros.cmake",
-    "dependencies": ["xtypes", "is-core", "is-ros2-rosidl"]
+    "dependencies": ["xtypes", "is-core", "is-mock"]
```

Comment:
The mock should not be a dependency of the System Handle itself, I suggest we delete `colcon.pkg` and let colcon decide the package build hierarchy based on the CMakeLists `find_package` instructions